_posts/2021-07-05-pcp-grafana.md

```diff
@@ -0,0 +1,106 @@
+author: pitti
+date: 2021-07-05
+category: howto
+tags: metrics pcp grafana
+---
+
+Debugging performance problems is a common troubleshooting activity. Monitoring the usage of CPU, memory, network, and other resources helps administrators to spot patterns when unusual resource usage happens.
+
+Cockpit can help you with this. Let's walk through!
+
+## Starting point
+
+Let's start with a pristine Fedora 34 system which has at least [Cockpit 247](https://cockpit-project.org/blog/cockpit-247.html). Everything here works in principle on CentOS/RHEL 8, Debian, and Ubuntu as well, but at the time of this writing these distributions still have an older Cockpit version. Let's make sure that Cockpit is installed, enabled, and opened in the firewall, as [documented on the Install page](https://cockpit-project.org/running.html):
+
+```
+# rpm -q cockpit
+cockpit-247-1.fc34.x86_64
+# systemctl enable --now cockpit.socket
+# firewall-cmd --permanent --add-service cockpit
+# firewall-cmd --reload
+```
+
+## Metrics in Cockpit for the local machine
+
+Then you can log into Cockpit on [https://localhost:9090](https://localhost:9090), or the corresponding host name of the system which you want to troubleshoot. The Overview page shows a very brief CPU/memory usage with a "View Details and history" link:
+
+![Overview usage card](/images/overview-card-usage.png)
+
+That link leads you to the Metrics page, which in a minimal installation only shows current metrics:
+
+![Metrics page without PCP](/images/metrics-page-no-pcp.png)
+
+For spotting performance problems in the past, we need to enable historical metrics, which are provided by [Performance Co-Pilot](https://pcp.io/), aka.  "PCP".
+
+The "Install cockpit-pcp" button installs the necessary support for this, and asks you to log in again (the new cockpit-pcp bridge cannot be used in an existing session). After that, the Metrics page will start to show historical metrics. Combined with Cockpit's other pages such as "Services", "Logs", or "Podman containers", this is a very powerful tool to find performance related problems:
+
+<iframe width="960" height="720" src="https://www.youtube.com/embed/8_NLoJBjay4?rel=0" frameborder="0" allowfullscreen></iframe>
+
+
+## Metrics of multiple machines
+
+But what if you want to monitor not just one machine, but your whole fleet?
+
+Cockpit used to have a "Dashboard" to see some basic resource usage of up to twenty machines, but this was very inflexible, inefficient, insecure, lacking important features such as alerting, and was bound to a running Cockpit session. That's why the Dashboard got removed in [Cockpit 234](https://cockpit-project.org/blog/cockpit-234.html).
+
+One of today's industry standard solutions for visualizing metrics is [Grafana](https://grafana.com/); it does not have above limitations. PCP offers a [Grafana plugin](https://grafana-pcp.readthedocs.io/en/latest/) to provide the machine's metrics to a local or remote Grafana installation.
```

Comment:
```suggestion
One of today's industry standard solutions for visualizing metrics is [Grafana](https://grafana.com/); it doesn't have above limitations. PCP offers a [Grafana plugin](https://grafana-pcp.readthedocs.io/en/latest/) to provide the machine's metrics to a local or remote Grafana installation.
```

Reply:
Out of interest, I thought contractions were informal, and should be avoided in written text?